src/components/Modals/WorldPermissionsModal/WorldPermissionsModal.tsx

```diff
@@ -0,0 +1,255 @@
+        setCollaboratorUserList(prev => [...prev, newAddress.toLowerCase()])
+      }
+      onGetProfilesRequest([newAddress])
+    },
+    [metadata.worldName, worldPermissions, newAddress, collaboratorUserList]
+  )
+
+  const handleUserPermissionList = useCallback(
+    (_event: React.MouseEvent, data: CheckboxProps & { worldPermissionName: WorldPermissionNames; wallet: string }) => {
+      const isValidAddress = isValid(data.wallet)
+      if (!isValidAddress) {
+        setErrorInvalidAddress(true)
+      }
+
+      if (
+        isValidAddress &&
+        worldPermissions?.[data.worldPermissionName].type === WorldPermissionType.AllowList &&
+        !(worldPermissions[data.worldPermissionName] as AllowListPermissionSetting).wallets.includes(data.wallet.toLowerCase())
+      ) {
+        setErrorInvalidAddress(false)
+        onPutWorldPermissionsRequest(metadata.worldName, data.worldPermissionName, WorldPermissionType.AllowList, data.wallet.toLowerCase())
+      }
+      if (
+        worldPermissions?.[data.worldPermissionName].type === WorldPermissionType.AllowList &&
+        (worldPermissions[data.worldPermissionName] as AllowListPermissionSetting).wallets.includes(data.wallet.toLowerCase())
+      ) {
+        setErrorInvalidAddress(false)
+        onDeleteWorldPermissionsRequest(
+          metadata.worldName,
+          data.worldPermissionName,
+          WorldPermissionType.AllowList,
+          data.wallet.toLowerCase()
+        )
+      }
+    },
+    [metadata.worldName, worldPermissions, newAddress]
+  )
+
+  const handleChangeAccessPermission = useCallback(
+    (_event: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => {
+      const nextPermissionType = !data.checked ? WorldPermissionType.AllowList : WorldPermissionType.Unrestricted
+      if (metadata.worldName && worldPermissions?.access.type !== nextPermissionType) {
+        onPostWorldPermissionsRequest(metadata.worldName, WorldPermissionNames.Access, nextPermissionType)
+      }
+    },
+    [metadata.worldName, worldPermissions]
+  )
+
+  const handleRemoveCollaborators = useCallback(
+    (_event: React.MouseEvent<HTMLButtonElement, MouseEvent>, data: ButtonProps & { wallet: string }) => {
+      if (
+        metadata.worldName &&
+        worldPermissions?.deployment.type === WorldPermissionType.AllowList &&
+        worldPermissions.deployment.wallets.includes(data.wallet)
+      ) {
+        onDeleteWorldPermissionsRequest(metadata.worldName, WorldPermissionNames.Deployment, WorldPermissionType.AllowList, data.wallet)
+      }
+
+      if (
+        metadata.worldName &&
+        worldPermissions?.streaming.type === WorldPermissionType.AllowList &&
+        worldPermissions.streaming.wallets.includes(data.wallet)
+      ) {
+        onDeleteWorldPermissionsRequest(metadata.worldName, WorldPermissionNames.Streaming, WorldPermissionType.AllowList, data.wallet)
+      }
+      setCollaboratorUserList(collaboratorUserList => {
```

Comment:
This is deleting the permission optimistically. It doesn't say anything if there's a failure while doing so. We should re-think the solution so when deleting a permission, the collaborators get updated from the redux state.
The same seems to be happening to all other of the requests.